include/ddc/kernels/splines/spline_evaluator_2d_batched.hpp

```diff
@@ -0,0 +1,410 @@
+            SplineBoundaryValue<bsplines_type2> const& left2_bc,
+            SplineBoundaryValue<bsplines_type2> const& right2_bc)
+        : m_spline_domain(spline_domain)
+    {
+    }
+
+    SplineEvaluator2DBatched(SplineEvaluator2DBatched const& x) = default;
+
+    SplineEvaluator2DBatched(SplineEvaluator2DBatched&& x) = default;
+
+    ~SplineEvaluator2DBatched() = default;
+
+    SplineEvaluator2DBatched& operator=(SplineEvaluator2DBatched const& x) = default;
+
+    SplineEvaluator2DBatched& operator=(SplineEvaluator2DBatched&& x) = default;
+
+
+
+    KOKKOS_FUNCTION spline_domain_type spline_domain() const noexcept
+    {
+        return m_spline_domain;
+    }
+
+    KOKKOS_FUNCTION bsplines_domain_type bsplines_domain() const noexcept // TODO : clarify name
+    {
+        return bsplines_domain_type(
+                ddc::discrete_space<bsplines_type1>().full_domain(),
+                ddc::discrete_space<bsplines_type2>().full_domain());
+    }
+
+    KOKKOS_FUNCTION batch_domain_type batch_domain() const noexcept
+    {
+        return ddc::remove_dims_of(spline_domain(), bsplines_domain());
+    }
+
+    template <class Layout, class... CoordsDims>
+    double operator()(
+            ddc::Coordinate<CoordsDims...> const& coord_eval,
+            ddc::ChunkSpan<double const, spline_domain_type, Layout, memory_space> const
+                    spline_coef) const
+    {
+        return eval(coord_eval, spline_coef);
+    }
+
+    template <class Layout1, class Layout2, class Layout3, class... CoordsDims>
+    void operator()(
+            ddc::ChunkSpan<double, vals_domain_type, Layout1, memory_space> const spline_eval,
+            ddc::ChunkSpan<
+                    ddc::Coordinate<CoordsDims...> const,
+                    vals_domain_type,
+                    Layout2,
+                    memory_space> const coords_eval,
+            ddc::ChunkSpan<double const, spline_domain_type, Layout3, memory_space> const
+                    spline_coef) const
+    {
+        interpolation_domain_type1 const interpolation_domain1(spline_eval.domain());
+        interpolation_domain_type2 const interpolation_domain2(spline_eval.domain());
+        ddc::for_each(
+                ddc::policies::policy(exec_space()),
+                batch_domain(),
+                KOKKOS_CLASS_LAMBDA(typename batch_domain_type::discrete_element_type const j) {
+                    const auto spline_eval_2D = spline_eval[j];
+                    const auto coords_eval_2D = coords_eval[j];
+                    const auto spline_coef_2D = spline_coef[j];
+                    for (auto const i1 : interpolation_domain1) {
+                        for (auto const i2 : interpolation_domain2) {
+                            spline_eval_2D(i1, i2) = eval(coords_eval_2D(i1, i2), spline_coef_2D);
+                        }
+                    }
+                });
+    }
+
+    template <class Layout, class... CoordsDims>
+    double deriv_dim_1(
```

Comment:
@tpadioleau do we keep the notation `deriv_dim_1 `or do we template a `deriv_dim `by dimensions on which we want to derive ?

Reply:
Both could be useful actually, depending if it is used in generic context or not

Reply:
Done in 871ab0f7f898141ccd9a2cf5ea34616de5c6ab66